### PR TITLE
Create Kasuraja Algorithm in C++

### LINKED ISSUE #488: Kasuraja Algorithm in C++
Hi, I would like to add Kasuraja algorithm for strongly connected components in a graph using c++.

### PR DESCRIPTION
**Pull request by BhanujaAggarwal**
Fixes #488 Kasuraja Algorithm in C++

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Kasuraja Algorithm in C++ 
- 

#### Languages Used:
- C++
- 

#### Files Added:
- Kasuraja_Algorithm.cpp
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by BhanujaAggarwal** [comment]
@jainaman224 Please review it

File: Goldmine/Goldmine_Problem.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+using namespace std; 
+
+const int MAX = 100; 
+
+int getMaxGold(int gold[][MAX], int m, int n) 
+{ 
+	int goldTable[m][n]; 
+	memset(goldTable, 0, sizeof(goldTable)); // Initialize all cells to 0
+
+	for (int col=n-1; col>=0; col--)
```
Comment (somya-kapoor) [inline Goldmine/Goldmine_Problem.cpp]: Please add space around the operators.

File: Goldmine/Goldmine_Problem.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	{ 
+		for (int row=0; row<m; row++)
```
Comment (somya-kapoor) [inline Goldmine/Goldmine_Problem.cpp]: Please add space around the operators.

File: Goldmine/Goldmine_Problem.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+{ 
+	int goldTable[m][n]; 
+	memset(goldTable, 0, sizeof(goldTable)); // Initialize all cells to 0
+
+	for (int col=n-1; col>=0; col--)
+	{ 
+		for (int row=0; row<m; row++) 
+		{ 
+			// Gold collected while going to right
+			int right = (col==n-1)? 0: goldTable[row][col+1];
```
Comment (somya-kapoor) [inline Goldmine/Goldmine_Problem.cpp]: Please add space around the operators.

File: Goldmine/Goldmine_Problem.cpp [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+	for (int col=n-1; col>=0; col--)
+	{ 
+		for (int row=0; row<m; row++) 
+		{ 
+			// Gold collected while going to right
+			int right = (col==n-1)? 0: goldTable[row][col+1]; 
+
+			// Gold collected while going to right up
+			int right_up = (row==0 || col==n-1)? 0:
```
Comment (somya-kapoor) [inline Goldmine/Goldmine_Problem.cpp]: Please add space around the operators.

File: Goldmine/Goldmine_Problem.cpp [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+		{ 
+			// Gold collected while going to right
+			int right = (col==n-1)? 0: goldTable[row][col+1]; 
+
+			// Gold collected while going to right up
+			int right_up = (row==0 || col==n-1)? 0: 
+							goldTable[row-1][col+1]; 
+
+			// Gold collected while going to right down
+			int right_down = (row==m-1 || col==n-1)? 0:
```
Comment (somya-kapoor) [inline Goldmine/Goldmine_Problem.cpp]: Please add space around the operators.

File: Goldmine/Goldmine_Problem.cpp [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+			// Max gold collected from above 3 paths
+			goldTable[row][col] = gold[row][col] + 
+							max(right, max(right_up, right_down)); 
+													
+		} 
+	} 
+
+	int res = goldTable[0][0]; 
+	for (int i=1; i<m; i++)
```
Comment (somya-kapoor) [inline Goldmine/Goldmine_Problem.cpp]: Please add space around the operators.

File: Goldmine/Goldmine_Problem.cpp [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+	int res = goldTable[0][0]; 
+	for (int i=1; i<m; i++) 
+		res = max(res, goldTable[i][0]);//max gold collected is max val in first column of all rows
+	return res; 
+} 
+
+int main() 
+{ 
+    int m,n;
```
Comment (somya-kapoor) [inline Goldmine/Goldmine_Problem.cpp]: Please add space after the comma.

File: Goldmine/Goldmine_Problem.cpp [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    cin>>m>>n;
```
Comment (somya-kapoor) [inline Goldmine/Goldmine_Problem.cpp]: Please add space around the operators.

File: Goldmine/Goldmine_Problem.cpp [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    int gold[MAX][MAX];
+    for(int i=0;i<m;i++){
```
Comment (somya-kapoor) [inline Goldmine/Goldmine_Problem.cpp]: Please add space around the operators.

File: Goldmine/Goldmine_Problem.cpp [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+        for(int j=0;j<n;j++){
```
Comment (somya-kapoor) [inline Goldmine/Goldmine_Problem.cpp]: Please add space around the operators.

File: Goldmine/Goldmine_Problem.cpp [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+            cin>>gold[i][j];
```
Comment (somya-kapoor) [inline Goldmine/Goldmine_Problem.cpp]: Please add space around the operators.

**Comment by somya-kapoor** [comment]
Please use 4 space indentation

**Comment by somya-kapoor** [comment]
@BhanujaAggarwal Are you working on it?

**Comment by somya-kapoor** [comment]
@BhanujaAggarwal Are you working on it?

**Comment by somya-kapoor** [comment]
Due to no activity for more than 7 days, I am closing this issue.

**Pull request closed without merging**